src/everest/optimizer/opt_model_transforms.py

```
@@ -443,3 +441,20 @@ def get_optimization_domain_transforms(
         "objective_scaler": objective_scaler,
         "constraint_scaler": constraint_scaler,
     }
+
+
+def _avg_functions(
+    realizations: NDArray[np.intc],
+    realization_weights: NDArray[np.float64],
+    functions: NDArray[np.float64],
+    error_msg: str,
+) -> NDArray[np.float64]:
+    realization_weights = np.tile(
+        realization_weights[realizations, np.newaxis], functions.shape[1]
+    )
+    realization_weights[np.isnan(functions)] = 0.0
+    rw_sum = np.sum(realization_weights, axis=0)
+    if np.any(rw_sum < np.finfo(np.float64).eps):
+        raise RuntimeError(error_msg)
```

is there a test that hits this? And why do we need to raise that error here when it is also potentially raised after? Will the condition here sometimes occur when the latter condition (at call site) does not?